src/components/select/__internal__/select-list/select-list.test.tsx

```diff
@@ -0,0 +1,1020 @@
+import React, { useRef, useState } from "react";
+import {
+  act,
+  fireEvent,
+  render,
+  screen,
+  waitFor,
+  within,
+} from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import mockDOMRect from "__spec_helper__/mock-dom-rect";
+
+import SelectList, { SelectListProps } from "./select-list.component";
+import Option from "../../option";
+import OptionRow from "../../option-row";
+
+beforeEach(() => {
+  jest.useFakeTimers();
+  // Mock non-zero dimensions for the scrollable container. To ensure react-virtual renders the list options correctly.
+  mockDOMRect(40, 100, "select-list-scrollable-container");
```

Comment:
@edleeks87 I noticed that we're using the `data-component` tag on an internal element, although the convention is to use it for marking the root container of Carbon components. Would you like me to update the `mockDOMRect` interface to accept a `data-role` identifier instead?

Since `mockDOMRect` is consumer-facing, we should consider any changes carefully.

Reply:
Good point, I think we should consider updating it soon but I don't think it's worth doing it in this PR as it's already got quite a bit to review etc